### PR TITLE
Updates Extools to 515 - 1606

### PR DESCRIPTION
**Pull request by JamieD1**

### CODE REVIEW

**Comment by Moja10** [comment]
I don't like it, please close?

**Comment by Moja10** [comment]
Please convert all the hexadecimal to human readable decimal system thanks xoxo

File: byond-extools/src/core/find_functions.cpp [hunk 1]
```diff
@@ -60,22 +79,20 @@ bool Core::find_functions()
...
 	FIND_OR_DIE(DisconnectClient1, "55 8B EC 6A FF 68 ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? 50 81 EC ?? ?? ?? ?? A1 ?? ?? ?? ?? 33 C5 89 45 F0 56 50 8D 45 F4 ?? ?? ?? ?? ?? ?? 6A 00 8D");
 	FIND_OR_DIE(DisconnectClient2, "55 8B EC 6A FF 68 ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? 50 51 53 56 57 A1 ?? ?? ?? ?? 33 C5 50 8D 45 F4 ?? ?? ?? ?? ?? ?? 8B 5D 08 ?? ?? ?? ?? ?? ?? ?? 0F 83");
-	FIND_OR_DIE(GetSocketHandleStruct, "55 8B EC FF 75 08 E8 ?? ?? ?? ?? 83 C4 04 85 C0 74 19 0F B7 40 02 8B ?? ?? ?? ?? ?? 50 E8 ?? ?? ?? ?? 85 C0 74 05 83 C0 74 5D C3 33 C0 5D C3");
-	FIND_OR_DIE(CallProcByName, "55 8B EC 83 EC 0C 53 8B 5D 10 8D 45 FF 56 8B 75 14 57 6A 01 50 FF 75 1C C6 45 FF 00 FF 75 18 6A 00 56 53");
-	FIND_OR_DIE(SendMaps, "55 8B EC 6A FF 68 ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? 50 81 EC ?? ?? ?? ?? A1 ?? ?? ?? ?? 33 C5 89 45 F0 53 56 57 50 8D 45 F4 ?? ?? ?? ?? ?? ?? A0 ?? ?? ?? ?? 84 C0 74 09 0F ?? ?? ?? ?? C0");
+	FIND_OR_DIE(GetSocketHandleStruct, "55 8B EC FF 75 08 E8 ?? ?? ?? ?? 83 C4 04 85 C0 74 19 0F B7 40 02 8B ?? ?? ?? ?? ?? 50 E8 ?? ?? ?? ?? 85 C0 74 05 83 C0");
+	FIND_OR_DIE(CallProcByName, "55 8B EC 81 EC 98 00 00 00 A1 ?? ?? ?? ?? 33 C5 89 45 FC 8B 55");
```
Comment (JamieD1) [inline byond-extools/src/core/find_functions.cpp]: Should be seperated

File: byond-extools/src/core/find_functions.cpp [hunk 2]
```diff
@@ -60,22 +79,20 @@ bool Core::find_functions()
...
+	FIND_OR_DIE(SendMaps, "55 8B EC 6A FF 68 C7 B0 31 10 64 A1 00 00 00 00 50 81 EC 58 07 00 00 A1 90 B3 38 10 33 C5 89 45 F0 53 56 57 50 8D 45 F4 64 A3 00 00 00 00 A0 98");
```
Comment (JamieD1) [inline byond-extools/src/core/find_functions.cpp]: here too

File: byond-extools/src/core/find_functions.cpp [hunk 3]
```diff
@@ -40,18 +40,37 @@ bool Core::find_functions()
...
+		FIND_OR_DIE(IsInContainer, "55 8B EC 8B 4D 10 83 EC 38 0F B6 C1 56 57 83 F8 53 0F 87 ?? ?? ?? ?? 0F B6 ?? ?? ?? ?? ?? FF 24 ?? ?? ?? ?? ?? 53 6A 0F FF 75 14 51 E8");
+		FIND_OR_DIE(DestroyDatum, "55 8B EC 8B 55 10 0F B6 C2 83 C0 FE 83 F8 1F 0F 87 ?? ?? ?? ?? 0F B6 ?? ?? ?? ?? ?? FF 24 ?? ?? ?? ?? ?? 8B 4D 14 3B ?? ?? ?? ?? ?? 0F 83 ?? ?? ?? ?? A1 ?? ?? ?? ?? 8B 04 88 85 C0 0F 84");
+	}
+
 	FIND_OR_DIE(TopicFloodCheck, "55 8B EC 6A FF 68 ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? 50 83 EC 2C 53 56 57 A1 ?? ?? ?? ?? 33 C5 50 8D 45 F4 ?? ?? ?? ?? ?? ?? FF 75 08 E8 ?? ?? ?? ??");
 	FIND_OR_DIE(DelDatum, "55 8B EC 8B 4D 08 3B 0D ?? ?? ?? ?? 73 55 A1 ?? ?? ?? ?? 8B 04 88 85 C0 74 49 8B 50 18 81 FA 00 00 00 70");
 	FIND_OR_DIE(Suspend, "55 8B EC 53 56 57 8B 7D 08 57 E8 ?? ?? ?? ?? 8B 1F 8B F0 8A 4F 69 83 C4 04 8B 56 18 88 4A 69 8B 4B 20 89 4E 20 8B 43 24 89 46 24 8B 45 0C C6 47 69 00 C7 43 ?? ?? ?? ?? ?? C7 43 ?? ?? ?? ?? ?? 8B 4E 18 89 41 04 F6 43 04 10");
 	FIND_OR_DIE(StartTiming, "55 8B EC 56 8B 75 ?? 56 80 4E ?? ??");
```
Comment (JamieD1) [inline byond-extools/src/core/find_functions.cpp:69]: this needs doing too

File: byond-extools/src/core/find_functions.cpp [hunk 4]
```diff
@@ -60,22 +79,20 @@ bool Core::find_functions()
...
-	FIND_OR_DIE(CallProcByName, "55 8B EC 83 EC 0C 53 8B 5D 10 8D 45 FF 56 8B 75 14 57 6A 01 50 FF 75 1C C6 45 FF 00 FF 75 18 6A 00 56 53");
-	FIND_OR_DIE(SendMaps, "55 8B EC 6A FF 68 ?? ?? ?? ?? ?? ?? ?? ?? ?? ?? 50 81 EC ?? ?? ?? ?? A1 ?? ?? ?? ?? 33 C5 89 45 F0 53 56 57 50 8D 45 F4 ?? ?? ?? ?? ?? ?? A0 ?? ?? ?? ?? 84 C0 74 09 0F ?? ?? ?? ?? C0");
+	FIND_OR_DIE(GetSocketHandleStruct, "55 8B EC FF 75 08 E8 ?? ?? ?? ?? 83 C4 04 85 C0 74 19 0F B7 40 02 8B ?? ?? ?? ?? ?? 50 E8 ?? ?? ?? ?? 85 C0 74 05 83 C0");
+	FIND_OR_DIE(CallProcByName, "55 8B EC 81 EC 98 00 00 00 A1 ?? ?? ?? ?? 33 C5 89 45 FC 8B 55");
+	FIND_OR_DIE(SendMaps, "55 8B EC 6A FF 68 C7 B0 31 10 64 A1 00 00 00 00 50 81 EC 58 07 00 00 A1 90 B3 38 10 33 C5 89 45 F0 53 56 57 50 8D 45 F4 64 A3 00 00 00 00 A0 98");
 	FIND_OR_DIE(GetGlobalByName, "55 8B EC 51 FF 35 ?? ?? ?? ?? E8 ?? ?? ?? ?? 83 C4 04 85 C0 75 26 FF 75 08 E8 ?? ?? ?? ?? FF 30 68 ?? ?? ?? ?? E8");
+	FIND_OR_DIE(DelDatum, "55 8b ec 8b 4d 08 3b 0d ?? ?? ?? ?? 73 55 a1 ?? ?? ?? ?? 8b 04 88 85 c0 74 49 8b 50 10 81 fa 00 00 00 70");
```
Comment (JamieD1) [inline byond-extools/src/core/find_functions.cpp]: double deldatum

File: byond-extools/src/core/find_functions.cpp [hunk 5]
```diff
@@ -87,51 +104,72 @@ bool Core::find_functions()
...
+	datum_pointer_table_length = *(unsigned int**)(datum_inc_function + 8);
+	datum_pointer_table = *(RawDatum****)(datum_inc_function + 15);
+
 	// the signature below is for a function that fetches a list via the real GetListPointerById and then checks if the list exists, runtiming otherwise
 	char* get_list_ptr_ptr = (char*)Pocket::Sigscan::FindPattern(BYONDCORE, "55 8B EC FF 75 08 E8 ?? ?? ?? ?? 83 C4 04 85 C0 75 13 68 ?? ?? ?? ?? E8 ?? ?? ?? ?? 83 C4 04 5D E9 ?? ?? ?? ?? 5D C3", 7);
 	GetListPointerById = (GetListPointerByIdPtr)(get_list_ptr_ptr + *(int*)get_list_ptr_ptr + 4);
 
-	current_execution_context_ptr = *(ExecutionContext * **)Pocket::Sigscan::FindPattern(BYONDCORE, "A1 ?? ?? ?? ?? 8D ?? ?? ?? ?? ?? 83 C4 08 89 48 28 8D ?? ?? ?? ?? ?? 89 48 2C 83 3D ?? ?? ?? ?? ?? 74 25 8B 00 FF 30 E8 ?? ?? ?? ?? 83 C4 04 FF 30 E8 ?? ?? ?? ?? 83 C4 04 FF 30 68 ?? ?? ?? ?? E8 ?? ?? ?? ?? 83 C4 08 66 ?? ?? ?? ?? ?? ?? A1 ?? ?? ?? ?? 75 28 A8 02 75 24 E8 ?? ?? ?? ?? 85 C0 75 09 50", 1);
+	current_execution_context_ptr = *(ExecutionContext * **)Pocket::Sigscan::FindPattern(BYONDCORE, "A1 ?? ?? ?? ?? FF 75 ?? 89 4D ?? 8B 4D ?? 8B 00 6A 00 52 6A 12 FF 70 ??", 1);
```
Comment (JamieD1) [inline byond-extools/src/core/find_functions.cpp]: Adam says this will break shit

File: byond-extools/src/core/find_functions.cpp (hunk 1 above)
Comment (JamieD1) [reply byond-extools/src/core/find_functions.cpp]: Was wrong before.

**Review by adamsong — CHANGES_REQUESTED** [review]
On windows
514: Hard crash, no log
515: Incorrect mappings for DestroyDatum, and SendMaps

**Pull request closed without merging**